modules/mono/glue/base_object_glue.cpp

```diff
@@ -81,13 +81,13 @@ void godot_icall_Object_Disposed(MonoObject *p_obj, Object *p_ptr) {
 void godot_icall_Reference_Disposed(MonoObject *p_obj, Object *p_ptr, MonoBoolean p_is_finalizer) {
 #ifdef DEBUG_ENABLED
 	CRASH_COND(p_ptr == nullptr);
-	// This is only called with Reference derived classes
-	CRASH_COND(!Object::cast_to<Reference>(p_ptr));
+	// This is only called with RefCounted derived classes
+	CRASH_COND(!Object::cast_to<RefCounted>(p_ptr));
 #endif
 
-	Reference *ref = static_cast<Reference *>(p_ptr);
+	RefCounted *rc = static_cast<RefCounted *>(p_ptr);
 
-	if (ref->get_script_instance()) {
+	if (rc->get_script_instance()) {
 		CSharpInstance *cs_instance = CAST_CSHARP_INSTANCE(ref->get_script_instance());
```

Comment:
These changes does not seem to be able to compile when enabling mono as ref looks have been renamed to rc

Reply:
#49562 :)